src/completion.ts

```diff
@@ -149,9 +154,9 @@ export function getCompletions(
     }
 
     let completion = new CompletionInfo(snippet, label, snippetRange, matchGroups);
-    if (snippet.automatic && snippetMatches) {
-      return completion;
-    } else if (prefixMatches) {
+    if (snippetMatches || prefixMatches) {
+      if(snippet.automatic) return [completion];
```

Comment:
We should be careful here, we don't want to trigger an automatic snippet expansion on a prefix match, that should only add the snippet to the completion list.

Reply:
The documentation deny it: "with the A flag snippets will activate as soon as their trigger matches, it is specially useful for regex snippets."
Why not making this flag available regardless of the matching mode? And is there any reason to differentiate both modes within the code? Maybe some change to the API as going to (label, regex (optional) defaults to label, description, flags) and changing the fixed string into a regex at the beginning could simplify the processing?